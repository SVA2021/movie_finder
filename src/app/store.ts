import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
