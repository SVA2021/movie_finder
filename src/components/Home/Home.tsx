import { Box, Typography } from "@mui/material";

const Home = () => {
	return (
		<Box sx={{
			height: '100px',
			display: 'flex',
			alignItems: 'center',
			justifyContent: 'center',
			bgcolor: 'action.hover',
			mt: 2,
		}}>
			<Typography variant="h1" component={'p'}>
				here will be a banner
			</Typography>
		</Box>
	)
}

export default Home;