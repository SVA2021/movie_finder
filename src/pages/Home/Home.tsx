import {Box, Link, Typography} from "@mui/material";
import {FC, useEffect} from 'react';
import {Link as RouterLink} from "react-router-dom";
import {useAppDispatch, useAppSelector} from "../../app/hooks";
import {SwipeListTemplate} from "../../components";
import {getTopListAsync, selectHomePage} from "../../features/movieFinder/movieFinderSlice";
import {TOP_FILMS_TYPE} from "../../features/movieFinder/movieFinderTypes";
import {getTopListPath, getTopListTitle} from "../../utils";

export const Home: FC<any> = () => {

	const dispatch = useAppDispatch();
	const HomePage = useAppSelector(selectHomePage);

	useEffect(() => {
		TOP_FILMS_TYPE.forEach((type) =>
			dispatch(getTopListAsync({type, page: 1}))
		)
	}, []);

	return (
		<Box pt={1}>
			{TOP_FILMS_TYPE.map((type, index) =>
				<Box m={1}  >
					<Typography
						ml={{lg: 15, xl: 20}}
						gutterBottom
						variant="h3"
						component={'h2'}
						textTransform={'capitalize'}
					>
						<Link
							component={RouterLink}
							underline={'hover'}
							to={getTopListPath(type)}
							sx={{
								color: 'common.white'
							}}
						>
							{getTopListTitle(type)}
						</Link>
					</Typography>
					<SwipeListTemplate key={index} data={HomePage[type]?.data ?? []} />
				</Box>
			)}
		</Box>
	)
};