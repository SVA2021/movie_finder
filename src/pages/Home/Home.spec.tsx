import {render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import {store} from "../../app/store";
import {Home} from "./Home";

describe('render Home', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('home')).toMatchSnapshot();
  });

})