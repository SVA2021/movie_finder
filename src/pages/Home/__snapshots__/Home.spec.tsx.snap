// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Home snapshot test 1`] = `
<div
  class="MuiBox-root css-zoser8"
  data-testid="home"
>
  <div
    class="MuiBox-root css-1ynyhby"
  >
    <h2
      class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1gwo2vl-MuiTypography-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-ezstlu-MuiTypography-root-MuiLink-root"
        href="/top/TOP_250_BEST_FILMS"
      >
        топ 250 фильмов кинопоиска
      </a>
    </h2>
    <div
      class="MuiGrid-root MuiGrid-container css-1nsfak8-MuiGrid-root"
      data-testid="swipe-list-template"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-h3 css-13mkr37-MuiTypography-root"
          data-testid="empty-data-box"
        >
          Нет данных
        </p>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-1ynyhby"
  >
    <h2
      class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1gwo2vl-MuiTypography-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-ezstlu-MuiTypography-root-MuiLink-root"
        href="/top/TOP_100_POPULAR_FILMS"
      >
        100 самых популярных фильмов
      </a>
    </h2>
    <div
      class="MuiGrid-root MuiGrid-container css-1nsfak8-MuiGrid-root"
      data-testid="swipe-list-template"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-h3 css-13mkr37-MuiTypography-root"
          data-testid="empty-data-box"
        >
          Нет данных
        </p>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-1ynyhby"
  >
    <h2
      class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1gwo2vl-MuiTypography-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-ezstlu-MuiTypography-root-MuiLink-root"
        href="/top/TOP_AWAIT_FILMS"
      >
        самые ожидаемые фильмы
      </a>
    </h2>
    <div
      class="MuiGrid-root MuiGrid-container css-1nsfak8-MuiGrid-root"
      data-testid="swipe-list-template"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-h3 css-13mkr37-MuiTypography-root"
          data-testid="empty-data-box"
        >
          Нет данных
        </p>
      </div>
    </div>
  </div>
</div>
`;
